varname/ignore.py

```diff
@@ -0,0 +1,343 @@
+
+IgnoreElemType = Union[
+    # module
+    ModuleType,
+    # filename of a module
+    str,
+    FunctionType,
+    # the module (filename) and qualname
+    # If module is None, then all qualname matches the 2nd element
+    # will be ignored. Used to ignore <listcomp/dictcomp/setcomp/genexpr>
+    # internally
+    Tuple[Optional[Union[ModuleType, str]], str],
+    # Function and number of its decorators
+    Tuple[FunctionType, int]
+]
+IgnoreType = Union[IgnoreElemType, List[IgnoreElemType]]
+
+MODULE_IGNORE_ID_NAME = '__varname_ignore_id__'
+
+def debug_ignore_frame(msg: str,
+                       frameinfo: Optional[inspect.FrameInfo] = None) -> None:
+    """Print the debug message for a given frame info object
+
+    Args:
+        msg: The debugging message
+        frameinfo: The FrameInfo object for the frame
+    """
+    from .utils import config
+    if not config.debug:
+        return
+    if frameinfo is not None:
+        msg = (f'{msg} [In {frameinfo.function!r} at '
+               f'{frameinfo.filename}:{frameinfo.lineno}]')
+    sys.stderr.write(f'[{__name__}] DEBUG: {msg}\n')
+
+class IgnoreElem(ABC):
+    """An element of the ignore list"""
+
+    def __init__(self, ignore: IgnoreElemType) -> None:
+        self.ignore = ignore
+
+    @abstractmethod
+    def match(self, frame_no: int, frameinfos: List[inspect.FrameInfo]) -> bool:
+        """Whether the frame matches the ignore element"""
+
+    @abstractmethod
+    def __repr__(self) -> str:
+        """Representation of the element"""
+
+class IgnoreModule(IgnoreElem):
+    """Ignore calls from a module or its submodules"""
+
+    def __init__(self, ignore: Union[ModuleType, str]) -> None:
+        super().__init__(ignore)
+        self.is_module = isinstance(ignore, ModuleType)
+        if self.is_module:
+            modfile = getattr(self.ignore, '__file__', None)
+            if not modfile or not path.isfile(modfile):
+                setattr(self.ignore, MODULE_IGNORE_ID_NAME,
+                        f'<varname-ignore-{id(self.ignore)}>')
+
+    def match(self, frame_no: int, frameinfos: List[inspect.FrameInfo]) -> bool:
+        frame = frameinfos[frame_no].frame
+
+        if self.is_module:
+            module = inspect.getmodule(frame)
+            if module:
+                return (module.__name__ == self.ignore.__name__ or
+                        module.__name__.startswith(f'{self.ignore.__name__}.'))
+
+            return (getattr(self.ignore, MODULE_IGNORE_ID_NAME, None) ==
+                    frame.f_globals.get(MODULE_IGNORE_ID_NAME, ''))
+
+        if path.isdir(self.ignore):
+            # ignore all calls from modules from a directory
+            # Used internally currently to ignore standard libraries
+            if not self.ignore.endswith('/'):
+                self.ignore = f'{self.ignore}/'
+            return frame.f_code.co_filename.startswith(self.ignore)
+
+        return frame.f_code.co_filename == self.ignore
```

Comment:
Sometimes `co_filename` is a relative path. I think you need something like `frame.f_code.co_filename == self.ignore or abspath(frame.f_code.co_filename) == abspath(self.ignore)`